Detectors/ITSMFT/ITS/workflow/src/TrackerSpec.cxx

```diff
@@ -144,32 +123,42 @@ void TrackerDPL::run(ProcessingContext& pc)
 
   LOG(INFO) << "ITSTracker pushed " << allTracks.size() << " tracks";
   pc.outputs().snapshot(Output{ "ITS", "TRACKS", 0, Lifetime::Timeframe }, allTracks);
-  pc.outputs().snapshot(Output{ "ITS", "TRACKSMCTR", 0, Lifetime::Timeframe }, allTrackLabels);
   pc.outputs().snapshot(Output{ "ITS", "ITSTrackROF", 0, Lifetime::Timeframe }, rofs);
-  pc.outputs().snapshot(Output{ "ITS", "ITSTrackMC2ROF", 0, Lifetime::Timeframe }, mc2rofs);
+  if (mIsMC) {
+    pc.outputs().snapshot(Output{ "ITS", "TRACKSMCTR", 0, Lifetime::Timeframe }, allTrackLabels);
+    pc.outputs().snapshot(Output{ "ITS", "ITSTrackMC2ROF", 0, Lifetime::Timeframe }, mc2rofs);
+  }
 
   mState = 2;
   pc.services().get<ControlService>().readyToQuit(false);
 }
 
-DataProcessorSpec getTrackerSpec()
+DataProcessorSpec getTrackerSpec(bool useMC)
 {
+  std::vector<InputSpec> inputs;
+  inputs.emplace_back("compClusters", "ITS", "COMPCLUSTERS", 0, Lifetime::Timeframe);
+  inputs.emplace_back("clusters", "ITS", "CLUSTERS", 0, Lifetime::Timeframe);
+  inputs.emplace_back("ROframes", "ITS", "ITSClusterROF", 0, Lifetime::Timeframe);
+
+  std::vector<OutputSpec> outputs;
+  outputs.emplace_back("ITS", "TRACKS", 0, Lifetime::Timeframe);
+  outputs.emplace_back("ITS", "ITSTrackROF", 0, Lifetime::Timeframe);
+
+  if (useMC) {
+    inputs.emplace_back("labels", "ITS", "CLUSTERSMCTR", 0, Lifetime::Timeframe);
+    inputs.emplace_back("MC2ROframes", "ITS", "ITSClusterMC2ROF", 0, Lifetime::Timeframe);
+    outputs.emplace_back("ITS", "TRACKSMCTR", 0, Lifetime::Timeframe);
+    outputs.emplace_back("ITS", "ITSTrackMC2ROF", 0, Lifetime::Timeframe);
+  }
+
   return DataProcessorSpec{
     "its-tracker",
-    Inputs{
-      InputSpec{ "compClusters", "ITS", "COMPCLUSTERS", 0, Lifetime::Timeframe },
-      InputSpec{ "clusters", "ITS", "CLUSTERS", 0, Lifetime::Timeframe },
-      InputSpec{ "labels", "ITS", "CLUSTERSMCTR", 0, Lifetime::Timeframe },
-      InputSpec{ "ROframes", "ITS", "ITSClusterROF", 0, Lifetime::Timeframe },
-      InputSpec{ "MC2ROframes", "ITS", "ITSClusterMC2ROF", 0, Lifetime::Timeframe } },
-    Outputs{
-      OutputSpec{ "ITS", "TRACKS", 0, Lifetime::Timeframe },
-      OutputSpec{ "ITS", "TRACKSMCTR", 0, Lifetime::Timeframe },
-      OutputSpec{ "ITS", "ITSTrackROF", 0, Lifetime::Timeframe },
-      OutputSpec{ "ITS", "ITSTrackMC2ROF", 0, Lifetime::Timeframe } },
-    AlgorithmSpec{ adaptFromTask<TrackerDPL>() },
+    inputs,
+    outputs,
+    AlgorithmSpec{ adaptFromTask<TrackerDPL>(useMC) },
     Options{
-      { "grp-file", VariantType::String, "o2sim_grp.root", { "Name of the output file" } },
+      { "grp-file", VariantType::String, "o2sim_grp.root", { "Name of the grp file" } },
+      { "nthreads", VariantType::Int, 1, { "Number of threads" } },
```

Comment:
@mpuccio  "nthreads" is not used anywhere in this piece of code.  This line could probably be removed.